### DISCUSSION TITLE
Forge Init - Weired Issue

### DISCUSSION THREAD

**Discussion by sudip358** [Cyfrin/foundry-full-course-cu#860] [General]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/16208424/a14833db-0402-4e23-bc42-12135b337724)

Not able to understand what i did wrong anyone. Can you please help me?

thanking you.

**Comment by 0xusmanf** [comment]
Can you please add more context to your question? so we can understand your issue and help you

**Reply by sudip358** [reply]
When i command forge install - - I get above output

**Reply by AztecWarlord** [reply]
#- If you are trying to install ```Foundry``` into your project use ```foundryup``` then hit enter. After the initial install, of course. Finally enter `forge unit`

**Reply by sudip358** [reply]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/16208424/2c0b6a30-4276-4cfe-9159-7158d33aa117)

getting same error after 

`/forge unit`

**Reply by 0xusmanf** [reply]
It's `forge init` not `unit`.

**Reply by AztecWarlord** [reply]
Ha yup! its ```forge init``` it was late for me last night!

**Reply by sudip358** [reply]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/16208424/65b03ac0-3c2f-49b8-b480-dafb1e89ddd8)
got the same error again

**Reply by AztecWarlord** [reply]
When you downloaded foundry using wsl did you restart your computer?

**Reply by 0xusmanf** [reply]
@theirrationalone @neftyr guys can you please help here, I am not very familiar with foundry installation issues.

@sudip358 Looks like foundry is not running or hasn't been installed correctly. Maybe try running `foundryup` before you run `forge init`.

**Reply by theirrationalone** [reply]
@sudip358 !, Dear you have to restart your machine and make sure to start wsl distro instance on restart. You have to look at bottom left corner to see if wsl instance is running. 

After that, try forge --version on bash terminal to confirm foundry. 

Read @usmanfarooq91 's last comment above.

**Reply by theirrationalone** [reply]
Heyyy @usmanfarooq91 !, He has no context, bound, closure for C, C++ Or more abstractly gnu gcc. @sudip358 !, You might forgot to install visual studio, that's why foundry sensed you might wanna work with binaries and how to modify program's binaries?? Modify the base program and that's forge.c that you're accessing.

Here is its core implementation [forge.c](https://github.com/KorfLab/SNAP/blob/master/forge.c) 

Mitigation... 


1. Delete wsl.
2. Install Node.js [node](https://nodejs.org/en) 
3. Install git [git](https://git-scm.com/downloads) 
4. Install visual studio  [vs](https://visualstudio.microsoft.com/) 
5. Install [wsl](https://learn.microsoft.com/en-us/windows/wsl/install) -read this carefully. 

6. Restart PC - > open visual studio -> install c/c++ binaries support

7. Open vscode not visual studio -> click on bottom left corner run wsl instance. 

8. Install foundry `curl -L https://foundry.paradigm.xyz | bash`.

9. Set source. 

10. Dev env. Ready 🧲 🚀🚀🚀🚀🚀🚀
